mlflow/server/handlers.py

```diff
@@ -756,8 +1053,63 @@ def _get_artifact_repo(run):
 @catch_mlflow_exception
 @_disable_if_artifacts_only
 def _log_batch():
+
+    def _assert_metrics_correctly_constructed(metrics):
+        for m in metrics:
+            print("This is a metric: ", m)
+            ## Key
+            k = m["key"]
+            _assert_required(k)
+            _assert_string(k)
+
+            ## Value
+            v = m["value"]
+            _assert_required(v)
+            _assert_float(v)
+
+            ## Timestamp
+            t = m["timestamp"]
+            _assert_required(t)
+            _assert_int(t)
+
+            ## Step
+            s = m["step"]
+            _assert_required(s)
+            _assert_int(s)
+
+    def _assert_params_correctly_constructed(params):
+        for p in params:
+            ## Key
+            k = p["key"]
+            _assert_required(k)
+            _assert_string(k)
+
+            ## Value
+            v = p["value"]
+            _assert_required(v)
+            _assert_string(v)
+
     _validate_batch_log_api_req(_get_request_json())
-    request_message = _get_request_message(LogBatch())
+    request_message = _get_request_message(
+        LogBatch(),
+        schema = {
+            "run_id": [
+                _assert_string,
+                _assert_required
+            ],
+            "metrics": [
+                _assert_array,
+                _assert_metrics_correctly_constructed
+            ],
+            "params": [
+                _assert_array,
+                _assert_params_correctly_constructed
+            ],
+            "tags": [
+                _assert_array
```

Comment:
Can we add `_assert_tags_fields_present`?

Reply:
does adding an `_assert_params_tags_correctly_constructed` (changed from `_assert_params_correctly_constructed`) work? The `RunTag` and `Param` data types look the same afaict